Automattic/wp-calypso
# Store: Display Stripe Connect Account details, if any

Adds the read-only connect account details view in support of #17002 

To test:

Test 1
The first test will be with no keys on the target site
Login to wp-admin on the target site, go to WooCommerce, Settings, Checkout, Stripe and clear the API keys and save.
Navigate to http://calypso.localhost:3000/store/settings/{site}, scroll down to Stripe and click on Manage.
Make sure you are prompted to create or sign in to your stripe account
Click on "I want to enter keys"
Make sure you are permitted to enter keys
Enter keys and hit Done to save

Test 2
This test will be with keys already set up on the target site
Navigate to http://calypso.localhost:3000/store/settings/{site}, scroll down to Stripe and click on Manage.
Make sure you see your keys
Make sure you are permitted to change other settings (e.g. ApplePay) and that they persist across a refresh.
Try the "I want to connect to Stripe" link and make sure you see the create/connect prompt.
Note that the full create/connect UX will be completed in a future PR - and that you cannot create or connect just yet.

Test 3
This test forces a connected but not activated account in props so you can see how it works
Modify payment-method-stripe.js by adding something like the following
```
	////////////////////////////////////////////////////////////////////////////
	// Temporary - will be removed in subsequent PR
	static defaultProps = {
		stripeConnectAccount: {
			connectedUserID: 'acct_14qyt6Alijdnw0EA', // e.g. acct_14qyt6Alijdnw0EA
			displayName: 'The Bar Company', // e.g. The Bar Company
			email: 'foo@bar.com', // e.g. foo@bar.com
			firstName: 'Foo',
			isActivated: false,
			lastName: 'Bar',
			logo: 'https://allendav.files.wordpress.com/2017/08/foo.png',
		}
	};
```
Navigate to http://calypso.localhost:3000/store/settings/{site}, scroll down to Stripe and click on Manage.
Ensure you see the connected account details, and that you are prompted to activate the account
Note: The disconnect link is not active and will be done in a future PR
Ensure you get Cancel and Done and either dismiss the dialog

Test 4
This test forces a connected and activated account in props so you can see how it works
Modify payment-method-stripe.js by adding something like the following
```
	////////////////////////////////////////////////////////////////////////////
	// Temporary - will be removed in subsequent PR
	static defaultProps = {
		stripeConnectAccount: {
			connectedUserID: 'acct_14qyt6Alijdnw0EA', // e.g. acct_14qyt6Alijdnw0EA
			displayName: 'The Bar Company', // e.g. The Bar Company
			email: 'foo@bar.com', // e.g. foo@bar.com
			firstName: 'Foo',
			isActivated: true,
			lastName: 'Bar',
			logo: 'https://allendav.files.wordpress.com/2017/08/foo.png',
		}
	};
```
Navigate to http://calypso.localhost:3000/store/settings/{site}, scroll down to Stripe and click on Manage.
Ensure you see the connected account details, and that the account says it is activated
Ensure you can change the method settings such as ApplePay and that those settings persist even after a forced refresh.

Test 5
This test forces the feature flag to false to ensure production is unchanged and working
Modify config/development so that `woocommerce/extension-settings-stripe-connect-flows` is false
Navigate to http://calypso.localhost:3000/store/settings/{site}, scroll down to Stripe and click on Manage.
Ensure that you see the original UX (with API keys and the like), that you are able to make changes, and that the changes persist after a refresh.
